src/components/MainSection.js

```diff
@@ -0,0 +1,111 @@
+import React from "react";
+import Styled from "styled-components";
+import MainSectionHeader from "./MainSectionHeader";
+import MainSectionResult from "./MainSectionResult";
+import ErrorBoundary from "./ErrorBoundary";
+// import WithLoading from "./WithLoading";
+
+//const ListWithLoading = WithLoading(MainSectionResult)
+const StyledFilterResult = Styled.h2`
+color: #fff;
+margin: 20px 0;
+font-size: 20px;
+`;
+const StyledFilterItems = Styled.div`
+display: flex;
+flex-wrap: wrap;
+`;
+const StyledBody = Styled.div`
+background-color: #232323;
+padding: 20px 50px;
+`;
+
+class MainSection extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      movies: [],
+    };
+  }
+
+  getData() {
+    setTimeout(() => {
```

Comment:
yeah, setTimeout could be used to emulate an asynchronous operation. It's just that it would be better to use Promise inside componentDidMount method, as if we're waiting to the server response and ready to process the incoming data